libafl/src/bolts/shmem.rs

```diff
@@ -209,7 +209,7 @@ pub trait ShMemProvider: Send + Clone + Default + Debug {
 #[derive(Debug, Clone)]
 pub struct RcShMem<T: ShMemProvider> {
     internal: ManuallyDrop<T::Mem>,
-    provider: Rc<RefCell<T>>,
+    provider: Rc<Mutex<RefCell<T>>>,
```

Comment:
Erm... that means you need an `Arc`, too, right?
Where is this needed? For the b2b stuff? should we use a new process there, too, to get around this `Mutex`?

Reply:
I found that there were all sorts of races with the `RcShMem`, adding the `Mutex` resolved them. I can add an `Arc` if you think it's necessary.